typescript/src/accounts/evm/toNetworkScopedEvmServerAccount.test.ts

```diff
@@ -0,0 +1,220 @@
+      id: "op-123",
+      network: "base",
+      targetAmount: "100",
+      targetCurrency: "eth",
+      status: "completed",
+      transactionHash: "0xtxhash",
+    }),
+    waitForFundOperationReceipt: async (): Promise<WaitForFundOperationResult> => ({
+      id: "op-123",
+      network: "base",
+      targetAmount: "100",
+      targetCurrency: "eth",
+      status: "completed",
+      transactionHash: "0xtxhash",
+    }),
+    quoteSwap: async (): Promise<AccountQuoteSwapResult> => ({
+      liquidityAvailable: true,
+      network: "base",
+      toToken: mockAddress,
+      fromToken: mockAddress,
+      fromAmount: 100n,
+      toAmount: 100n,
+      minToAmount: 100n,
+      blockNumber: 100n,
+      fees: {
+        gasFee: {
+          amount: 100n,
+          token: mockAddress,
+        },
+      },
+      issues: {
+        simulationIncomplete: false,
+      },
+      execute: async () => ({
+        transactionHash: "0xtxhash",
+      }),
+    }),
+    swap: async () => ({
+      transactionHash: "0xtxhash",
+    }),
+    transfer: async () => ({
+      transactionHash: "0xtxhash",
+    }),
+    sendTransaction: async () => ({
+      transactionHash: "0xtxhash",
+    }),
+  });
+
+  it("should create a network-scoped account with all properties", () => {
+    const mockAccount = createMockEvmServerAccount();
+    const result = toNetworkScopedEvmServerAccount(mockApiClient, {
+      account: mockAccount,
+      network: mockNetwork,
+    });
+
+    expect(result).toEqual({
+      address: mockAddress,
+      network: mockNetwork,
+      name: mockName,
+      type: "evm-server",
+      policies: mockPolicies,
+      signMessage: mockAccount.signMessage,
+      sign: mockAccount.sign,
+      signTransaction: mockAccount.signTransaction,
+      signTypedData: mockAccount.signTypedData,
+    });
+  });
+
+  it("should preserve all signing functions from the original account", async () => {
+    const mockAccount = createMockEvmServerAccount();
+    const result = toNetworkScopedEvmServerAccount(mockApiClient, {
+      account: mockAccount,
+      network: mockNetwork,
+    });
+
+    // Test that all signing functions are preserved and work as expected
+    const message = { message: "test" };
+    const hash = "0xhash";
+    const transaction: TransactionSerializable = {
+      to: mockAddress,
+      value: 100n,
+      chainId: 1,
+    };
+    const typedData = {
+      domain: { name: "Test" },
+      types: { Test: [{ name: "test", type: "string" }] },
+      primaryType: "Test",
+      message: { test: "test" },
+    };
+
+    expect(await result.signMessage(message)).toBe("0xsignature");
+    expect(await result.sign({ hash })).toBe("0xsignature");
+    expect(await result.signTransaction(transaction)).toBe("0xsignedTx");
+    expect(await result.signTypedData(typedData)).toBe("0xsignature");
+  });
+
+  it("should handle accounts without optional properties", () => {
+    const mockAccount: EvmServerAccount = {
```

Comment:
We're redefining a lot of things here that are already defined in the return value of `createMockEvmServerAccount` – I would simplify by doing something like:

```ts
let mockAccount = createMockEvmServerAccount();
mockAccount = {
  ...mockAccount,
  policies: undefined,
  name: undefined
}
```

I also am somewhat questioning the value of this test, and would be ok with removing it altogether